### PR TITLE
Backup command

### LINKED ISSUE #23: Requesting /backup ability for gangmembers
I hope this one is valid :P
Anyway I'm going to make a pull request.

### PR DESCRIPTION
**Pull request by AmirSavand**
## Backup command
Calls all gang members for backup via `SendClientMessage()`

### Fixes
Fixes #23 
Also add and use `IsPlayerInAnyGang` for commands, less code equals more awesomeness.

### Note
It's not tested or compiled at all, if code style or anything else is out of ordinary, update the pull :wink:.

### CODE REVIEW

File: SS_GANGSYSTEM.pwn [hunk 1]
```diff
@@ -1647,6 +1648,30 @@ CMD:ghelp(playerid)
...
+*/
+CMD:backup(playerid)
+{
+    // Is a gang member
+    if (!IsPlayerInAnyGang(playerid)) return 1;
+
+    foreach (new p: SS_Player)
+    {
+        // Is gangemate
+        if (!strcmp(GInfo[playerid][gangname],GInfo[p][gangname])) continue;
```
Comment (AmirSavand) [inline SS_GANGSYSTEM.pwn]: I think there's a problem in the condition, I gotta make sure it's right, if not will update the pull.

Comment (amirm3hdi) [inline SS_GANGSYSTEM.pwn]: I think there's a problem in the condition, I gotta make sure it's right, if not will update the pull.

Comment (amirm3hdi) [inline SS_GANGSYSTEM.pwn]: Fixes

**Comment by AmirSavand** [comment]
@Sreyas-Sreelal @Konstantinos-Sk I think it's ready, just need you to test it out.

File: README.md [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
-* Gang Zone will locked for certain time given by user      
-* Capturing gang zone gives 10 score to the gang              
-* In game dynamic gang zone creator                           
-* On entering the zone zone info will be displayed to player  
-	                                                           
+*   /gcp        - to enter gang control panel
+*   /creategang - to create new gang
+*   /gangtag    - to add tag to your gang
+*   /gwar       - to challenge other gang members for a gang war
+*   /backup     - to request backup
```
Comment (AmirSavand) [inline README.md:11]: If `/backup` doesn't belong to version 1 section, change it.

**Comment by Konstantinos-Sk** [comment]
The name _IsPlayerInAnyGang_ is misleading, I'd expect from it a true/false value to return. What I think you meant is a PlayerInGangCheck with a #define pattern which is a simply find and replace.

About the command you forgot the player loop and yes, the condition is wrong. You need to check if strcmp returns anything but 0 to skip the iteration.

**Comment by AmirSavand** [comment]
@Konstantinos-Sk I changed the name to `CheckGangMemberShip()` also I already fixed the `strcmp()` condition.

I'm done now, update the pull by your desire.

**Comment by Konstantinos-Sk** [comment]
You forgot the two arguments in format (name, id) and and seeing your code, I completely forgot there is SendGangMessage function so the loop is unnecesary, sorry about that.

By the way, wait for Sreyas about the pull request. I took over the mysql+streamer version and haven't touched the "master" branch at all.

**Comment by AmirSavand** [comment]
@Konstantinos-Sk Yea yea, I saw it too, I used the existing function, but forgot to remove the loop haha

**Comment by Sreyas-Sreelal** [comment]
Seems legitimate and thanks

**Comment by Konstantinos-Sk** [comment]
CheckGangMembership will never return 0 so it will always proceed. Basically what I have thought of this was:
```pawn
#define CheckGangMembership(%0) if (!GInfo[%0][gangmember]) return SendClientMessage(%0,-1,""RED"ERROR:"GREY"You are not a Gang Member")
```

and in commands:
```pawn
CMD:...(playerid, params[])
{
    CheckGangMembership(playerid);

    // code..
    return 1;
}
```

**Comment by Sreyas-Sreelal** [comment]
The use of macros extensively would create a dilemma in future that's why i try to minimize the use of macros but the above statement has a point. But what you say about this
```
CheckGangMembership(playerid)
  +{
  +    if (!GInfo[playerid][gangmember]) 
            {
                 SendClientMessage(playerid,-1,""RED"ERROR:"GREY"You are not a Gang Member");
                 return 0;
             }
  +    return 1;
  +}
```
Which one is better to implement ? i didn't see any difference though but seems more readable to others

**Comment by Konstantinos-Sk** [comment]
It's only on compile-time so I don't see any problem. It's still better than calling a function just to send an error message.

**Comment by Sreyas-Sreelal** [comment]
If that's your opinion then it will be the best as you always give right suggestions :).

**Comment by AmirSavand** [comment]
Awesome guys.

**Pull request merged**